chain/jsonrpc-primitives/src/types/transactions.rs

```diff
@@ -0,0 +1,119 @@
+use near_primitives::borsh::BorshDeserialize;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+#[derive(Debug, Clone)]
+pub struct RpcTransactionRequest {
+    pub signed_transaction: near_primitives::transaction::SignedTransaction,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum RpcTransactionError {
+    // TxStatusError
+    // NetworkClientResponses::InvalidTx
+    //NetworkClientResponses::NoResponse
+    // DoesNotTrackShard,
+    // Ban { ban_reason: ReasonForBan },
+    // ServerError (Timeout, Closed)
+    #[error("The node does not track shard")]
+    DoesNotTrackShard,
+    #[error("An error happened during transaction execution: {context:?}")]
+    InvalidTransaction { context: near_primitives::errors::InvalidTxError },
+    #[error(
+        "Requested transaction with hash {requested_transaction_hash} is not found on the node"
+    )]
+    UnknownTransaction { requested_transaction_hash: near_primitives::hash::CryptoHash },
+    #[error("The node reached its limits. Try again later. More details: {debug_info}")]
+    InternalError { debug_info: String },
+    #[error("Timeout")]
+    TimeoutError,
+    // NOTE: Currently, the underlying errors are too broad, and while we tried to handle
+    // expected cases, we cannot statically guarantee that no other errors will be returned
+    // in the future.
+    // TODO #3851: Remove this variant once we can exhaustively match all the underlying errors
+    #[error("It is a bug if you receive this error type, please, report this incident: https://github.com/near/nearcore/issues/new/choose. Details: {error_message}")]
+    Unreachable { error_message: String },
+}
+
+// TODO: ask if we want to create From<NetworkClientResponse> for it
+#[derive(Serialize, Deserialize, Debug)]
+pub struct RpcTransactionResponse {
+    #[serde(flatten)]
+    pub final_execution_outcome: near_primitives::views::FinalExecutionOutcomeViewEnum,
+}
+
+#[derive(Debug)]
+pub struct RpcTransactionCheckValidResponse;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct RpcBroadcastTxSyncResponse {
+    pub transaction_hash: String,
+    pub is_routed: bool,
```

Comment:
I thought we disabled routing?

Reply:
Only in query. I haven't heard of disabling routing for broadcasting transactions.

Reply:
@frol @khorolets it is effective disabled because we disabled receipt responses https://github.com/near/nearcore/blob/ab4f0c1c481b503e725798e994a58964ed6bc49b/chain/client/src/view_client.rs#L984. After you route the transaction, you won't be able to get its status

Reply:
@bowenwang1996 and we don't try to get its status here, as you can see, if `RequestRouted` is returned, we give up on waiting and return it to the user. Do you mean that we cannot even get `RequestRouted` result?

Reply:
I see. I think I misremembered how it works